testdata/baselines/reference/submodule/compiler/contravariantOnlyInferenceFromAnnotatedFunctionJs.errors.txt

```diff
@@ -0,0 +1,39 @@
+index.js(2,28): error TS2304: Cannot find name 'B'.
+index.js(2,42): error TS2304: Cannot find name 'A'.
+index.js(2,48): error TS2304: Cannot find name 'B'.
+index.js(2,67): error TS2304: Cannot find name 'B'.
+
+
+==== index.js (4 errors) ====
+    /**
+     * @typedef {{ [K in keyof B]: { fn: (a: A, b: B) => void; thing: B[K]; } }} Funcs
+                               ~

```

Comment:
my best guess is that the something set the type expression parents to the synthetic type alias, inconsistent with other type expressions, and the binder change now makes it consistent with the others. I have a branch that sets `Host` on typedef children which should fix this.

Edit:actually it may be that all  type aliases were working by  mistake?

Reply:
actually it may be that *all*  typedefs were setting their type's parent to the synthetic node. Not in the reparser, though, so through the binder's existing code.